apps/api/src/app/webhooks/usecases/get-webhook-portal-token/get-webhook-portal-token.usecase.ts

```diff
@@ -0,0 +1,53 @@
+import { BadRequestException, Inject, Injectable, NotFoundException, Scope } from '@nestjs/common';
+import { EnvironmentRepository } from '@novu/dal';
+import { LogDecorator, SvixClient } from '@novu/application-generic';
+
+import { GetWebhookPortalTokenCommand } from './get-webhook-portal-token.command';
+import { GetWebhookPortalTokenResponseDto } from '../../dtos/get-webhook-portal-token-response.dto';
+
+const LOG_CONTEXT = 'GetWebhookPortalTokenUsecase';
+
+@Injectable()
+export class GetWebhookPortalTokenUsecase {
+  constructor(
+    private environmentRepository: EnvironmentRepository,
+    @Inject('SVIX_CLIENT') private svix: SvixClient
+  ) {}
+
+  @LogDecorator()
+  async execute(command: GetWebhookPortalTokenCommand): Promise<GetWebhookPortalTokenResponseDto> {
+    if (!this.svix) {
+      throw new BadRequestException('Webhook system is not enabled');
+    }
+
+    const environment = await this.environmentRepository.findOne({
+      _id: command.environmentId,
+      _organizationId: command.organizationId,
+    });
+
+    if (!environment) {
+      throw new NotFoundException(
+        `Environment not found for id ${command.environmentId} and organization ${command.organizationId}`
+      );
+    }
+
+    try {
+      const svixResponse = await this.svix.authentication.appPortalAccess(
+        `${command.organizationId}-${command.environmentId}`,
+        {}
+      );
+
+      return {
+        url: svixResponse.url,
+        token: svixResponse.token,
+        appId: `${command.organizationId}-${command.environmentId}`,
```

Comment:
maybe we should create a function to centralize this appid creation.

```suggestion
        appId: `o=${command.organizationId}:e=${command.environmentId}`,
```